### PR TITLE
Script rewrites

### PR DESCRIPTION
**Pull request by tmarcu**

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -3,11 +3,13 @@ dist_bin_SCRIPTS = \
 	create-update.sh \
 	build-chroots.sh \
 	init-mix.sh \
-	init-versions.sh \
 	update-bundles.sh \
 	pack-maker.sh \
 	superpack-maker.sh
 
+instdir = ${datadir}/mixer-tools
```
Comment (phmccarty) [inline Makefile.am]: should be `$(datadir)` instead of `${datadir}`

File: add-rpms.sh [hunk 2]
```diff
@@ -1,77 +1,73 @@
 #!/bin/bash
+source /usr/share/mixer-tools/helpers
```
Comment (phmccarty) [inline add-rpms.sh]: Note that you are installing this file to `$(datadir)/helpers`, so if mixer-tools is built with a datadir different than `/usr/share`, the file sourcing here will fail.

One solution would be to make the scripts templates instead, and substitute the correct values at configure time. The line in the template would look like `source @datadir@/mixer-tools/helpers`.

File: add-rpms.sh [hunk 3]
```diff
@@ -1,77 +1,73 @@
...
-  ret=$(file "$rpm")
-  echo "$ret" | grep --quiet ": RPM"
-  if [[ $? -ne 0 ]] ; then
-    echo "ERROR $rpm IS NOT VALID"
-  else
-    echo "Copying $rpm"
-    cp "$rpm" "$REPODIR"
-  fi
-done
+ret=$(file "$rpm")
```
Comment (phmccarty) [inline add-rpms.sh]: indentation was removed for this line...

File: create-update.sh [hunk 4]
```diff
@@ -89,58 +73,35 @@ fi
...
 
 # step 4: hardlink relevant dirs
 sudo -E "hardlink" -f "$STATE_DIR/image/$MIXVER"/
 
 # step 5: update latest version
 if [ $NOPUBLISH -eq 0 ]; then
-	sudo cp "$PWD/.mix-version" "$STATE_DIR/image/LAST_VER"
-	sudo cp "$PWD/.mix-version" "$STATE_DIR/www/version/format$FORMAT/latest"
+	sudo -E echo "$MIXVER" > "$STATE_DIR/image/LAST_VER"
+	sudo -E echo "$MIXVER" > "$STATE_DIR/www/version/format$FORMAT/latest"
```
Comment (phmccarty) [inline create-update.sh]: `echo` might be a shell builtin, in which case I would expect these two commands to fail. I would suggest using `sh -c ...` instead.

File: init-mix.sh [hunk 5]
```diff
@@ -1,13 +1,12 @@
-#!/bin/bash
-
+#!/bin/bash -x
```
Comment (phmccarty) [inline init-mix.sh]: shell tracing on by default?

File: update-bundles.sh [hunk 6]
```diff
@@ -1,48 +1,48 @@
-#!/bin/bash
+#!/bin/bash -x
```
Comment (phmccarty) [inline update-bundles.sh]: shell tracing on by default?

File: helpers [hunk 7]
```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
```
Comment (tmarcu) [inline helpers]: Yes ty for pointing that out... should have been clear in my editor, I'll fix.

File: init-mix.sh (hunk 5 above)
Comment (tmarcu) [inline init-mix.sh]: Debug artifact. removed.

File: update-bundles.sh (hunk 6 above)
Comment (tmarcu) [inline update-bundles.sh]: Same as previous, removed.

File: Makefile.am (hunk 1 above)
Comment (phmccarty) [inline Makefile.am]: it's for style consistency

File: Makefile.am [hunk 8]
```diff
@@ -3,11 +3,13 @@ dist_bin_SCRIPTS = \
 	create-update.sh \
 	build-chroots.sh \
 	init-mix.sh \
-	init-versions.sh \
```
Comment (tmarcu) [inline Makefile.am]: No longer needed. The builder.conf hosts the values, and I will adjust the scripts to auto-increment or set the next mix value when starting to build the version. Imho we shouldn't need a whole file to edit a version #.

Comment (phmccarty) [inline Makefile.am]: I'm okay with that approach. The removal was something that glaringly stood out when reviewing this patch, as there was no mention of it in the commit message.

Comment (tmarcu) [inline Makefile.am]: Good point, I will amend the commit message, it should be stated.

**Pull request merged**